pkg/minikube/registry/drvs/docker/docker.go

```diff
@@ -130,6 +132,75 @@ func status() registry.State {
 	return checkNeedsImprovement()
 }
 
+func checkDockerVersion(o string) registry.State {
+	parts := strings.SplitN(o, "-", 2)
+	if len(parts) != 2 {
+		return registry.State{
+			Reason:    "PROVIDER_DOCKER_VERSION_PARSING_FAILED",
+			Error:     errors.Errorf("expected version string format is \"{{.Server.Os}}-{{.Server.Version}}\". but got %s", o),
+			Installed: true,
+			Healthy:   false,
+			Doc:       docURL,
+		}
+	}
+
+	if parts[0] == "windows" {
+		return registry.State{
+			Reason:    "PROVIDER_DOCKER_WINDOWS_CONTAINERS",
+			Error:     oci.ErrWindowsContainers,
+			Installed: true,
+			Healthy:   false,
+			Fix:       "Change container type to \"linux\" in Docker Desktop settings",
+			Doc:       docURL + "#verify-docker-container-type-is-linux",
+		}
+	}
+
+	p := strings.SplitN(parts[1], ".", 3)
+	switch l := len(p); l {
+	case 2:
+		p = append(p, "0") // patch version not found
+	case 3:
+		//remove postfix string for unstable(test/nightly) channel. https://docs.docker.com/engine/install/
+		p[2] = strings.SplitN(p[2], "-", 2)[0]
+	default:
+		return registry.State{
+			Reason:    "PROVIDER_DOCKER_VERSION_PARSING_FAILED",
+			Error:     errors.Errorf("expected version format is \"<year>.<month>.{patch}\". but got %s", parts[1]),
+			Installed: true,
+			Healthy:   false,
+			Doc:       docURL,
+		}
+	}
+
+	for i, s := range p {
+		k, err := strconv.Atoi(s)
+		if err != nil {
+			return registry.State{
+				Reason:    "PROVIDER_DOCKER_VERSION_PARSING_FAILED",
+				Error:     errors.Wrap(err, "docker version"),
+				Installed: true,
+				Healthy:   false,
+				Doc:       docURL,
+			}
+		}
+
+		if k > minDockerVersion[i] {
+			return registry.State{Installed: true, Healthy: true, Error: nil}
+		} else if k < minDockerVersion[i] {
+			return registry.State{
+				Reason:           "PROVIDER_DOCKER_VERSION_LOW",
+				Error:            oci.ErrMinDockerVersion,
+				Installed:        true,
+				Healthy:          false,
+				NeedsImprovement: true,
+				Fix:              fmt.Sprintf("Upgrade docker server to a newer version (Minimum supproted version is %2d.%2d.%2d)", minDockerVersion[0], minDockerVersion[1], minDockerVersion[2]),
```

Comment:
no need to say docker server, you should also use this function
https://github.com/medyagh/minikube/blob/72443005764661ff12dafedee6c0afed4829b379/pkg/minikube/driver/driver.go#L193

driver.FullName(ds.Name)

so it will give the right name for the user

simmilar to 
https://github.com/medyagh/minikube/blob/72443005764661ff12dafedee6c0afed4829b379/cmd/minikube/cmd/start.go#L716